otherlibs/unix/unixsupport_unix.c

```diff
@@ -321,6 +321,12 @@ void caml_uerror(const char *cmdname, value cmdarg)
   caml_unix_error(errno, cmdname, cmdarg);
 }
 
+CAMLprim void caml_unix_uerror(value msg, value cmdarg) {
+  CAMLparam0();
+  caml_uerror(String_val(msg), cmdarg);
```

Comment:
Looking at the source of `caml_uerror`, it looks like this is safe (the GC will not be triggered before `String_val(msg)` is copied), but it is subtle and fragile (a change in `caml_uerror` can silently cause a segfault here). Can we find a more robust way of writing this code?